File: BusTicketingAI.Domain/Entity/Bus.cs
namespace BusTicketingAI.Domain.Entity;

public class Bus
{
    public int Id { get; set; }
    public int CompanyId { get; set; }
    public string PlateNumber { get; set; } = string.Empty;
    public int SeatCapacity { get; set; }

    public BusCompany Company { get; set; } = null!;
}

File: BusTicketingAI.Client/src/app/core/models/bus.ts
export interface Bus {
    id: number;
    plateNumber: string;
    seatCapacity: number;
    companyId: number;
}

export interface CreateTripRequest {
    busId: number;
    originTerminalId: number;
    destinationTerminalId: number;
    departureTime: string;
    basePrice: number;
    estimatedDuration: number;
}

export interface UpdateTripTimeRequest {
    newDepartureTime: string;
}